posts/2020-09-29-microprofile-openapi-beta-200011.adoc

```diff
@@ -0,0 +1,247 @@
+author_github: https://github.com/jakub-pomykala
+seo-title: MicroProfile OpenAPI 2.0 plus new features and loads of improvements for Jakarta EE 9 in Open Liberty 20.0.0.11-beta - OpenLiberty.io
+seo-description: Open Liberty 20.0.0.11-beta comes with MicroProfile OpenAPI 2.0 which allows users to generate documentation directly from API's, plus two new features for Jakarta EE 9.
+blog_description: Open Liberty 20.0.0.11-beta comes with MicroProfile OpenAPI 2.0 which allows users to generate documentation directly from API's, plus two new features for Jakarta EE 9.
+---
+= New features and loads of improvements for Jakarta EE 9 plus MicroProfile OpenAPI 2.0 in Open Liberty 20.0.0.11-beta
+Jakub Pomykala <https://github.com/jakub-pomykala>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+
+In this beta release the All Beta Features package contains our new MicroProfile OpenAPI 2.0 which allows you to generate documentation directly from API's. In Jakarta EE 9 we have two features and also include improvements to features including many moving up to the latest RC's of API's. 
+
+We have two beta packages for link:{url-about}[Open Liberty]:
+
+* <<allbeta, All Beta Features>>: a new, larger package that contains all Open Liberty beta and GA features and functions.
+* <<jakarta, Jakarta EE 9 Beta Features>>: a lightweight package that contains only the Jakarta EE 9 features.
+
+This means that you can now try out our in-development Open Liberty features by just adding the relevant coordinates to your build tools (either the All Beta Features package or the Jakarta EE 9 Beta Features package, depending on what you want to try).
+
+
+If you try either package, <<feedback, let us know what you think>>.
+[#allbeta]
+== All Beta Features package
+
+As well as the Jakarta EE 9 features listed above, the new All Beta Features package contains our new MicroProfile OpenAPI 2.0 feature which have been available in our link:/downloads/#development_builds[nightly builds] for a few weeks but are now easy to install from Maven Central too.
+
+[#openAPI]
+=== MicroProfile OpenAPI 2.0 
+
+The link:https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md[OpenAPI v3 specification] defines a standard, language-agnostic, interface for describing REST APIs which allows documentation to be generated from the APIs themselves. The link:https://download.eclipse.org/microprofile/microprofile-open-api-2.0-RC3/microprofile-openapi-spec.html[MicroProfile OpenAPI specification] provides a unified Java API for the OpenAPI v3 specification which allows Java developers to generate OpenAPI v3 documents from their JAX-RS applications.
+MicroProfile OpenAPI 2.0 of the MicroProfile OpenAPI specification introduces some new annotations that simplify the process of generating OpenAPI documentation for your REST APIs. It also introduces a new MicroProfile Config property prefix that can be used to define the schema for Java classes that you do not have the source code for.
```

Comment:
"MicroProfile OpenAPI 2.0 of the MicroProfile OpenAPI" looks like a real typo?

Reply:
Done